app/src/test/java/com/fstyle/structure_android/screen/main/MainPresenterTest.java

```diff
@@ -39,58 +37,53 @@
     Validator mValidator;
     @Mock
     CustomCompositeSubscription mSubscription;
+    @Mock
+    UserRepository mUserRepository;
 
-    private UserRepository mUserRepository;
     private MainPresenter mMainPresenter;
 
     @Before
     public void setUp() throws Exception {
         MockitoAnnotations.initMocks(this);
-        RxAndroidPlugins.getInstance().registerSchedulersHook(new RxAndroidSchedulersHook() {
-            @Override
-            public Scheduler getMainThreadScheduler() {
-                return Schedulers.immediate();
-            }
-        });
-        mUserRepository = new UserRepository(mLocalDataSource, mRemoteDataSource);
-        mMainPresenter = new MainPresenter(mUserRepository, mValidator, mSubscription);
+        mMainPresenter = new MainPresenter(mUserRepository, mValidator, mSubscription,
+                new ImmediateSchedulerProvider());
         mMainPresenter.setViewModel(mViewModel);
     }
 
-    @After
-    public void tearDown() {
-        RxAndroidPlugins.getInstance().reset();
-    }
-
     @Test
-    public void searchUsers() throws Exception {
+    public void searchUsers_ValidDataInput_ShowListUser() {
```

Comment:
ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.5.1/pmd-java/rules/java/naming.html#MethodNamingConventions